### PR TITLE
Issue #SB-15180 added functionality for sending sms based on template

### PR DESCRIPTION
**Pull request by Hari-stackroute**

### CODE REVIEW

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java [hunk 1]
```diff
@@ -29,26 +29,28 @@ public void onReceive(Request request) throws Throwable {
...
+      sendOTPViaEmail(key, otp, userId, template);
     } else if (JsonKey.PHONE.equalsIgnoreCase(type)
         || JsonKey.PREV_USED_PHONE.equalsIgnoreCase(type) || JsonKey.RECOVERY_PHONE.equalsIgnoreCase(type)) {
-      sendOTPViaSMS(key, otp);
+      sendOTPViaSMS(key, otp, template);
     }
   }
 
-  private void sendOTPViaEmail(String key, String otp, String otpType) {
+  private void sendOTPViaEmail(String key, String otp, String otpType, String template) {
```
Comment (manzarul) [inline actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: is OtpType required here?

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java [hunk 2]
```diff
@@ -58,10 +60,11 @@ private void sendOTPViaEmail(String key, String otp, String otpType) {
     tellToAnother(emailRequest);
   }
 
-  private void sendOTPViaSMS(String key, String otp) {
+  private void sendOTPViaSMS(String key, String otp, String template) throws Exception {
```
Comment (manzarul) [inline actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: Don't throw Exception

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/service/OTPService.java [hunk 3]
```diff
@@ -26,26 +33,26 @@ public void deleteOtp(String type,String key){
...
+   try {
+    Properties props = new Properties();
+    props.put("resource.loader", "class");
+    props.put(
+            "class.resource.loader.class",
+            "org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader");
+
+    VelocityEngine ve = new VelocityEngine();
+    ve.init(props);
+    Template t = ve.getTemplate("/"+templateFile);
```
Comment (manzarul) [inline actors/common/src/main/java/org/sunbird/learner/actors/otp/service/OTPService.java]: Read template from DB only , no need to put inside file.

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java [hunk 4]
```diff
@@ -29,26 +29,28 @@ public void onReceive(Request request) throws Throwable {
...
     }
   }
 
-  private void sendOTPViaEmail(String key, String otp, String otpType) {
+  private void sendOTPViaEmail(String key, String otp, String otpType, String template) {
     Map<String, Object> emailTemplateMap = new HashMap<>();
     emailTemplateMap.put(JsonKey.EMAIL, key);
     emailTemplateMap.put(JsonKey.OTP, otp);
     emailTemplateMap.put(JsonKey.OTP_EXPIRATION_IN_MINUTES, OTPUtil.getOTPExpirationInMinutes());
+    emailTemplateMap.put(JsonKey.TEMPLATEID, template);
```
Comment (manzarul) [inline actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: If provided template is incorrect then it should throw data error.

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java (hunk 1 above)
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: yes.

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java [hunk 5]
```diff
@@ -29,26 +29,28 @@ public void onReceive(Request request) throws Throwable {
...
     if (JsonKey.EMAIL.equalsIgnoreCase(type) || JsonKey.PREV_USED_EMAIL.equalsIgnoreCase(type)|| JsonKey.RECOVERY_EMAIL.equalsIgnoreCase(type)) {
       String userId = (String) request.get(JsonKey.USER_ID);
       ProjectLogger.log(
           "SendOTPActor:sendOTP userId found for send otp " + userId, LoggerEnum.INFO.name());
-      sendOTPViaEmail(key, otp, userId);
+      sendOTPViaEmail(key, otp, userId, template);
     } else if (JsonKey.PHONE.equalsIgnoreCase(type)
         || JsonKey.PREV_USED_PHONE.equalsIgnoreCase(type) || JsonKey.RECOVERY_PHONE.equalsIgnoreCase(type)) {
-      sendOTPViaSMS(key, otp);
+      sendOTPViaSMS(key, otp, template);
```
[the remark this answers is not in the document]
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: Taken care in controller.

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/SchedulerManager.java [hunk 6]
```diff
@@ -270,6 +270,7 @@ private void scheduleShadowUser(String identifier) {
             .build();
     String shadowUserTime =
         PropertiesCache.getInstance().getProperty("quartz_shadow_user_migration_timer");
+    shadowUserTime = "0 0/10 * 1/1 * ? *";
```
[the remark this answers is not in the document]
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/common/quartz/scheduler/SchedulerManager.java]: Removed this change

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java [hunk 7]
```diff
@@ -29,26 +29,28 @@ public void onReceive(Request request) throws Throwable {
     }
   }
 
-  private void sendOTP(Request request) {
+  private void sendOTP(Request request) throws Exception {
```
[the remark this answers is not in the document]
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: removed this.

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java (hunk 2 above)
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: Removed this.

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/service/OTPService.java (hunk 3 above)
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/service/OTPService.java]: Now sms templates are reading from DB.

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java (hunk 4 above)
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/SendOTPActor.java]: Taken care in controller.

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/service/OTPService.java (hunk 3 above)
Comment (manzarul) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/service/OTPService.java]: is cql taken care for other env as well.

**Pull request merged**